apps/zero-runtime-vite-app/src/App.tsx

```diff
@@ -171,6 +168,7 @@ export function App({ isRed }: AppProps) {
           <span>Hello2</span>
         </div>
       </div>
+      <Box as="div">Hello</Box>
```

Comment:
Shouldn't it be `<Box component="div">`?

Reply:
`as` comes from emotion, but we can decide on the convention. Anyways, this is not the final "Box" component. Just an implementation in the demo app.

Reply:
We can certainly do this.

Reply:
Is there an opportunity to move to "as", with "component" as a supported but undocumented prop to make it a non-breaking change?

https://github.com/mui/material-ui/pull/40863 illustres the nuance between the `as` and `component` prop. It might be possible, but require a behavior change, so likely a BC. Here is why: 

- Developers using styled() are used to be able to use the `as` prop, so unlikely something we can change.
- It's common to wrap a Material UI component with a styled version of it, using styled() (in v4, we would do this with withStyles). But at that point the as prop replaces the component wrapped with styled() not the React host DOM element.
- So either we use two different props (as + components) or we make it so that styled only handle the as prop if he can see a React host DOM element, which is not emotion, styled components behavior.